src/withStyles.jsx

```diff
@@ -40,24 +42,25 @@ export function withStyles(
   const styleDef = styleFn && ThemedStyleSheet.create(styleFn);
   const BaseClass = baseClass(pureComponent);
 
+  // As some components will depend on previous styles in the component tree, we
+  // provide the option of flushing the buffered styles (i.e. to a style tag)
+  // **before** the rendering cycle begins.
+  //
+  // The interfaces provide the optional "flush" method which is run in turn by
+  // ThemedStyleSheet.flush.
+  const flushIfNecessary = flushBefore ? ThemedStyleSheet.flush : NOOP;
+
   return function withStylesHOC(WrappedComponent) {
+    const isClass = !!(WrappedComponent.prototype && WrappedComponent.prototype.isReactComponent);
```

Comment:
That's true, but I doubt that will be a problem. If something weird sets it, this will have the same behavior as before.